include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -79,7 +79,250 @@ namespace LinearAlgebra
         return a;
       }
 
+#ifdef DEAL_II_USE_KOKKOS_BACKEND
+      template <typename Number, typename MemorySpaceType>
+      struct la_parallel_vector_templates_functions
+      {
+        using size_type = types::global_dof_index;
+
+        static void
+        resize_val(
+          const types::global_dof_index new_alloc_size,
+          types::global_dof_index &     allocated_size,
+          ::dealii::MemorySpace::MemorySpaceData<Number, MemorySpaceType> &data,
+          const MPI_Comm &comm_shared)
+        {
+          if (comm_shared == MPI_COMM_SELF)
+            {
+              allocated_size = new_alloc_size;
+              Kokkos::realloc(data.values, new_alloc_size);
+
+              if (std::is_same<MemorySpaceType, Kokkos::HostSpace>::value)
+                {
+                  data.values_sm = {
+                    ArrayView<const Number>(data.data(), new_alloc_size)};
+                }
+            }
+          else
+            {
+#  ifdef DEAL_II_WITH_MPI
+#    if DEAL_II_MPI_VERSION_GTE(3, 0)
+              allocated_size = new_alloc_size;
+
+              const unsigned int size_sm =
+                Utilities::MPI::n_mpi_processes(comm_shared);
+              const unsigned int rank_sm =
+                Utilities::MPI::this_mpi_process(comm_shared);
+
+              MPI_Win mpi_window;
+              Number *data_this;
+
+
+              std::vector<Number *> others(size_sm);
+
+              MPI_Info info;
+              int      ierr = MPI_Info_create(&info);
+              AssertThrowMPI(ierr);
+
+              ierr = MPI_Info_set(info, "alloc_shared_noncontig", "true");
```

Comment:
This is the same code from `AlignedVector::replicate...()`, right? We should really break this out into a tool function -- but I'm not asking for this here.

Reply:
Though this code seems to have noticeably fewer comments than what's in `AlignedVector::replicate...()` :-)

Reply:
I don't know . This code is mainly a copy/paste of the non-kokkos backend. The code has already a lot of  `#ifdef` so I thought it would be easier to read if I clearly separate the Kokkos backend from the non-kokkos backend.